pyrefly/lib/playground.rs

```diff
@@ -177,31 +223,131 @@ impl Playground {
+
+        for (filename, content) in &files {
+            let module_name =
+                ModuleName::from_str(filename.strip_suffix(".py").unwrap_or(filename));
+
+            let module_path = PathBuf::from(format!("{}.py", module_name.as_str()));
+            let memory_path = ModulePath::memory(module_path.clone());
+
+            module_mappings.insert(module_name, memory_path.dupe());
+
+            let handle = Handle::new(module_name, memory_path, self.sys_info.dupe());
+            self.handles.insert(filename.clone(), handle);
+            file_contents.push((module_path, Some(Arc::new(content.clone()))));
+        }
+
+        let source_db = PlaygroundSourceDatabase::new(module_mappings, self.sys_info.dupe());
+        config.source_db = Some(ArcId::new(Box::new(source_db) as Box<dyn SourceDatabase>));
+
+        config.configure();
+        let config = ArcId::new(config);
+        let new_config_finder = ConfigFinder::new_constant(config.dupe());
+
+        self.state = State::new(new_config_finder);
+        self.config_finder = ConfigFinder::new_constant(config.dupe());
+
+        if let Some(sandbox_handle) = self.handles.get("sandbox.py") {
+            self.handle = sandbox_handle.dupe();
+        } else if let Some((_, first_handle)) = self.handles.iter().next() {
+            self.handle = first_handle.dupe();
+        }
+
+        let mut transaction = self
+            .state
+            .new_committable_transaction(Require::Exports, None);
+        transaction.as_mut().set_memory(file_contents);
+
+        let handle_reqs: Vec<(Handle, Require)> = self
+            .handles
+            .values()
+            .map(|handle| (handle.dupe(), Require::Everything))
+            .collect();
+
         self.state
-            .transaction()
-            .get_errors([&self.handle])
-            .collect_errors()
-            .shown
-            .into_map(|e| {
-                let range = e.display_range();
-                Diagnostic {
-                    start_line: range.start.line.get() as i32,
-                    start_col: range.start.column.get() as i32,
-                    end_line: range.end.line.get() as i32,
-                    end_col: range.end.column.get() as i32,
-                    message_header: e.msg_header().to_owned(),
-                    message_details: e.msg_details().unwrap_or("").to_owned(),
-                    kind: e.error_kind().to_name().to_owned(),
-                    // Severity values defined here: https://microsoft.github.io/monaco-editor/typedoc/enums/MarkerSeverity.html
-                    severity: match e.error_kind().default_severity() {
-                        Severity::Error => 8,
-                        Severity::Warn => 4,
-                        Severity::Info => 2,
-                        Severity::Ignore => 1, // Ignored errors shouldn't be in `CollectedErrors.shown`
-                    },
+            .run_with_committing_transaction(transaction, &handle_reqs);
```

Comment:
need to rebase here, we changed the shape of this.

Instead of handle_reqs being a list of pairs of handle + requires, we now pass a list of handles, and a single requires that applies to all of them.